### PR TITLE
docs(json): complete documentation

### PR DESCRIPTION
**Pull request by sylc**
Towards https://github.com/denoland/deno_std/issues/3764

`deno doc --lint json/mod.ts` has no errors.

### CODE REVIEW

File: json/json_stringify_stream.ts [hunk 1]
```diff
@@ -82,6 +82,9 @@ export interface StringifyStreamOptions {
  * ```
  */
 export class JsonStringifyStream extends TransformStream<unknown, string> {
+  /**
+   * Construct new instance
+   */
```
Comment (crowlKats) [inline json/json_stringify_stream.ts:87]: This should have documentaiton about the options that can be provided.

Comment (sylc) [reply json/json_stringify_stream.ts:92]: done

Comment (iuioiua) [reply json/json_stringify_stream.ts:92]: @crowlKats, `StringifyStreamOptions` is fully documented above. To clarify, did you mean to provide example snippets where the constructor is used in different ways? Is that required if we already have examples on the class above?

**Review by iuioiua — APPROVED** [review]
I've made a bunch of tweaks. PTAL, @sylc. These were done as I was figuring out how this should all be done. Regarding class constructors, their argument types were documented, which suffices. Examples are done on the classes themselves, rather than the constructors.

Thank you for this.

**Pull request merged**